peer/network_test.go

```diff
@@ -132,6 +132,54 @@ func TestRequestAnyRequestsRoutingAndResponse(t *testing.T) {
 	assert.Equal(t, totalCalls, int(atomic.LoadUint32(&callNum)))
 }
 
+func TestAppRequestOnCtxCancellation(t *testing.T) {
+	var net Network
+	codecManager := buildCodec(t, HelloRequest{}, HelloResponse{})
+	crossChainCodecManager := buildCodec(t, ExampleCrossChainRequest{}, ExampleCrossChainResponse{})
+
+	callNum := uint32(0)
+	senderWg := &sync.WaitGroup{}
+	sender := testAppSender{
+		sendAppRequestFn: func(nodes set.Set[ids.NodeID], requestID uint32, requestBytes []byte) error {
```

Comment:
Are these functions actually doing anything? Seems like they could be empty. The test passes when they are, and making them empty would allow some code removal.

Reply:
Not really, can remove

Reply:
Actually we need to define them since we use the functions in `SendAppRequest` etc

Reply:
Otherwise we would be accessing a nil pointer